### PR TITLE
Add resource limits to nvmeof configuration

### PR DESCRIPTION
**Pull request by gbregman**
mgr/cephadm: Add resource limits to nvmeof configuration
python-common/ceph/deployment: Add resource limits to nvmeof configuration

    Fixes https://tracker.ceph.com/issues/68967

    Signed-off-by: Gil Bregman <gbregman@il.ibm.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by gbregman** [comment]
@adk3798 and @VallariAg had to re-open PR #60757 under this PR because of some rebase issues. The code is about the same. Just added some more tests in the validation.

**Review by adk3798 — APPROVED** [review]

File: src/python-common/ceph/deployment/service_spec.py [hunk 1]
```diff
@@ -1613,6 +1625,36 @@ def validate(self) -> None:
         ):
             raise SpecValidationError("Log file directory backups can't be negative")
 
+        if (self.max_hosts_per_namespace and self.max_hosts_per_namespace < 0):
+            raise SpecValidationError("Max hosts per namespace can't be negative")
+
+        if (self.max_namespaces_with_netmask and self.max_namespaces_with_netmask < 0):
+            raise SpecValidationError("Max namespaces with netmask can't be negative")
+
+        if type(self.max_subsystems) != int:
```
Comment (idryomov) [inline src/python-common/ceph/deployment/service_spec.py:1634]: This appears to be causing `make check` failures:
```
lint: commands[0] /home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common> flake8 ceph
ceph/deployment/service_spec.py:1642:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ceph/deployment/service_spec.py:1648:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ceph/deployment/service_spec.py:1654:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ceph/deployment/service_spec.py:1660:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lint: exit 1 (1.52 seconds) /home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common> flake8 ceph pid=45070
lint: FAIL ✖ in 26.1 seconds
```

Comment (gbregman) [reply src/python-common/ceph/deployment/service_spec.py:1634]: @idryomov I didn't get these when I committed. Anyway, I'm about to make another change there so I'll change this too.

Comment (phlogistonjohn) [reply src/python-common/ceph/deployment/service_spec.py:1634]: Possibly PR #60587 - as a newer flake8 was needed to support newer python versions such as 3.12.

Comment (phlogistonjohn) [reply src/python-common/ceph/deployment/service_spec.py:1634]: let me know if you need help fixing these

Comment (idryomov) [reply src/python-common/ceph/deployment/service_spec.py:1634]: Already fixed in #60818 ;)

**Pull request merged**